storage/replica.go

```diff
@@ -1122,6 +1122,17 @@ func (r *Replica) Send(
 	return br, pErr
 }
 
+func (r *Replica) checkBatchRange(ba roachpb.BatchRequest) error {
+	rspan, err := keys.Range(ba)
+	if err != nil {
+		return err
+	}
+	return r.checkCmdHeader(roachpb.Span{
+		Key:    rspan.Key.AsRawKey(),
```

Comment:
It's not dangerous, but a little hacky. Pulling on that thread a little, it seems that you should always pass the resolved keys into `checkCmdHeader` (the range descriptor has a `ContainsKeyRange` method which takes `RKey`).

Besides, isn't the only other callsite `addAdminCmd` where you can also just pass the batch instead of the request?